src/Suggestor/Suggestor.ts

```diff
@@ -496,14 +497,37 @@ export function onlySuggestIfBracketOpen(fn: SuggestionBuilder, brackets: [strin
  * and false value otherwise.
  *
  * This checks for simple pre-conditions:
+ *  - Does the parent editor explicitly request it?
  *  - Is the global filter (if set) in the line?
- *  - Is the line a task line (with a checkbox)
+ *  - Is the line a task line (with a checkbox)?
  * @param line
- * @param cursorPosition - the cursor position, when 0 is presumed to mean 'at the start of the line'.
- *                          See 'ch' in https://docs.obsidian.md/Reference/TypeScript+API/EditorPosition
+ * @param cursor - the cursor position, when ch is 0 it is presumed to mean 'at the start of the line'.
+ *                          See https://docs.obsidian.md/Reference/TypeScript+API/EditorPosition
+ * @param editor - the editor instance to which the suggest belongs
  */
-export function canSuggestForLine(line: string, cursorPosition: number) {
-    return GlobalFilter.getInstance().includedIn(line) && cursorIsInTaskLineDescription(line, cursorPosition);
+export function canSuggestForLine(line: string, cursor: EditorPosition, editor: Editor) {
+    const lineHasGlobalFilter = GlobalFilter.getInstance().includedIn(line);
+    const didEditorRequest = editorIsRequestingSuggest(editor, cursor, lineHasGlobalFilter);
+
+    if (typeof didEditorRequest === 'boolean') return didEditorRequest;
+    return lineHasGlobalFilter && cursorIsInTaskLineDescription(line, cursor.ch);
+}
+
+/**
+ * Return
+ * - true if the parent editor is explicitly requesting that the suggest be displayed
+ * - false if it is requesting that it be hidden
+ * - undefined if the parent editor wants to defer to the default behavior
+ *
+ * @param editor - the editor instance to which the suggest belongs
+ * @param cursor - the cursor position
+ */
+function editorIsRequestingSuggest(
+    editor: Editor,
+    cursor: EditorPosition,
+    lineHasGlobalFilter: boolean,
+): boolean | undefined {
+    return (editor as any)?.editorComponent?.showTasksPluginAutoSuggest?.(cursor, editor, lineHasGlobalFilter);
```

Comment:
@mgmeyers just a comment here for my knowledge. I'm curious why the `editorComponent` is needed? In Kanban plugin code (https://github.com/mgmeyers/obsidian-kanban/blob/5fa792b9c2157390fe493f0feed6f0bc9be72910/src/components/Editor/MarkdownEditor.tsx#L100-L106) I don't see that being used, what am I missing here? Why not calling just `(editor as any)?.showTasksPluginAutoSuggest?.(cursor, editor, lineHasGlobalFilter);`?

Reply:
So that's bad naming on my part 😆. The `Editor` I have in the kanban plugin is actually an `EditorComponent`. 

The unofficial API for obsidian's `Editor` that gets passed into the suggest is:

```typescript
class Editor {
    editorComponent: EditorComponent;
}
```

Reply:
Wait, so this is Kanban-specific thing? Could you please render that more generic so that other plugins would use this too in Obsidian way?

Reply:
I'm pretty sure that this implementation works and this is not my concern. What I'd like to be sure about here is interoperability with Obsidian API that would permit this PR to scale to other plugins as well =) Could you please help me understand if this is the case or not?

Reply:
No, this isn't specific to Kanban, though Kanban is the first community plugin to implement a custom editor. It's important to note, though, that implementing a custom editor requires extending undocumented functionality from Obsidian. 

Obsidian has two classes relevant to this discussion: `Editor` and `EditorComponent`. The `Editor` class has an `EditorComponent` as a property:

```typescript
class Editor {
    editorComponent: EditorComponent;
}
```

`EditorComponent` houses the Codemirror editor instance whereas `Editor` is an abstraction that provides various helper functions for interacting with Codemirror. The `EditorComponent` class is not exported through Obsidian's public API, but `Editor` is.

Generally, Obsidian passes around the `Editor` class instead of `EditorComponent`. For example, Obsidian's `Editor` class is passed in as the second parameter to `EditorSuggestor.onTrigger()`.

The Kanban editor extends the `EditorComponent` class and therefore lives under `editor.editorComponent` by default.

Reply:
To respond to your second comment. This could indeed scale to other plugins. Perhaps the risk here, though, is that it relies on undocumented functionality and therefore could stop working at some point if Obsidian decides to change the internal API. The worst case scenario is that `editorIsRequestingSuggest` would always return `undefined` and require updating to match internal API changes. It would make sense to me if this feels too much of a risk, and I would be happy to revert this PR in that case.

Reply:
I'm pretty ok to leave this and not revert back since it benefits Kanban plugin (Which I didn't use before because it was not supporting Tasks).

But I'd like to explore how this whole ecosystem works to provide better APIs to other plugins like Kanban. You seem to understand this part quite well, would you care to have a live discussion about this, just the 2 of us? I could write you a mail mentioned on the site from your profile. Please let me know =)

Reply:
Yes, I'd be happy to chat!

Reply:
Oks, sent you a mail)